File: dotnet/Multiplayer/Rooms/RoomListResponse.cs
using Michitai.Multiplayer.Errors;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Michitai.Multiplayer.Rooms
{
    /// <summary>
    /// Response containing a list of available game rooms.
    /// </summary>
    /// <typeparam name="T">The type to deserialize room rules into.</typeparam>
    public class RoomListResponse<T> : ApiResponse<ERoomListError> where T : class, new()
    {
        /// <summary>
        /// List of available game rooms with their details.
        /// </summary>
        public List<RoomShort<T>> Rooms { get; set; } = new();
    }
}

File: java/src/main/java/com/michitai/multiplayer/rooms/RoomListResponse.java
package com.michitai.multiplayer.rooms;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.michitai.multiplayer.ApiResponse;

import java.util.List;

/**
 * Response containing a list of available game rooms.
 *
 * @param <T> The type to deserialize room rules into.
 */
public class RoomListResponse<T> extends ApiResponse {
    @JsonProperty("rooms")
    private List<RoomShort<T>> rooms;

    @JsonProperty("count")
    private int count;

    public List<RoomShort<T>> getRooms() {
        return rooms;
    }

    public void setRooms(List<RoomShort<T>> rooms) {
        this.rooms = rooms;
    }

    public int getCount() {
        return count;
    }

    public void setCount(int count) {
        this.count = count;
    }
}
